packages/grid/_modules_/grid/hooks/features/core/useGridState.ts

```diff
@@ -2,30 +2,89 @@ import * as React from 'react';
 import { GRID_STATE_CHANGE } from '../../../constants/eventsConstants';
 import { GridApiRef } from '../../../models/api/gridApiRef';
 import { GridStateChangeParams } from '../../../models/params/gridStateChangeParams';
+import { isDeepEqual } from '../../../utils/utils';
 import { GridState } from './gridState';
 import { useGridApi } from './useGridApi';
 
 export const useGridState = (
   apiRef: GridApiRef,
 ): [GridState, (stateUpdaterFn: (oldState: GridState) => GridState) => boolean, () => void] => {
   useGridApi(apiRef);
+
   const forceUpdate = React.useCallback(
     () => apiRef.current.forceUpdate(() => apiRef.current.state),
     [apiRef],
   );
+
   const setGridState = React.useCallback(
     (stateUpdaterFn: (oldState: GridState) => GridState) => {
       const newState = stateUpdaterFn(apiRef.current.state);
       const hasChanged = apiRef.current.state !== newState;
+      let shouldUpdate = true;
```

Comment:
Shouldn't it begin as `false` to update the state as little as possible? If I have an `controlState.propOnChange` set and the new model is equal to `controlState.propModel` it publishes the state change event.

Reply:
by default it should let the state update unless it is controlled. This is not to control the number of state updates